c-dependencies/js-compute-runtime/builtins/backend.h

```diff
@@ -1,8 +1,273 @@
+   * Cipher suites using ephemeral ECDH key agreement, excluding anonymous cipher suites.
+   * Same as "kEECDH:-AECDH"
+   */
+  inline static std::string EECDH = "EECDH";
+  /**
+   * Cipher suitesusing ECDH key exchange, including anonymous, ephemeral and fixed ECDH.
+   */
+  inline static std::string ECDH = "ECDH";
+  /**
+   * Cipher suites using ephemeral ECDH key agreement, including anonymous cipher suites.
+   */
+  inline static std::string kECDHE = "kECDHE";
+  /**
+   * Cipher suites using authenticated ephemeral ECDH key agreement
+   */
+  inline static std::string ECDHE = "ECDHE";
+  /**
+   * Cipher suites using 128 bit AES.
+   */
+  inline static std::string AES128 = "AES128";
+  /**
+   * Cipher suites using 256 bit AES.
+   */
+  inline static std::string AES256 = "AES256";
+  /**
+   * Cipher suites using either 128 or 256 bit AES.
+   */
+  inline static std::string AES = "AES";
+  /**
+   * AES in Galois Counter Mode (GCM): these cipher suites are only supported in TLS v1.2.
+   */
+  inline static std::string AESGCM = "AESGCM";
+  /**
+   * Cipher suites using CHACHA20.
+   */
+  inline static std::string CHACHA20 = "CHACHA20";
+  /**
+   * Cipher suites using triple DES.
+   */
+  inline static std::string TRIPLE_DES = "3DES";
+  /**
+   * Cipher suites using SHA1.
+   */
+  inline static std::string SHA1 = "SHA1";
+  /**
+   * Cipher suites using SHA1.
+   */
+  inline static std::string SHA = "SHA";
+  /**
+   * Cipher suites using SHA256.
+   */
+  inline static std::string SHA256 = "SHA256";
+  /**
+   * Cipher suites using SHA384.
+   */
+  inline static std::string SHA384 = "SHA384";
+
+  // The content of the default list is determined at compile time and normally corresponds to
+  // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+  inline static std::string DEFAULT = "DEFAULT";
+  // The ciphers included in ALL, but not enabled by default.
+  inline static std::string COMPLEMENTOFDEFAULT = "COMPLEMENTOFDEFAULT";
+
+  inline static std::string ALL = "ALL";
+
+  static void init(std::map<std::string, std::vector<Cipher>> *aliases);
+
+  static void moveToEnd(std::map<std::string, std::vector<Cipher>> *aliases,
+                        std::vector<Cipher> *ciphers, std::string cipher);
+
+  static void moveToEnd(std::vector<Cipher> *ciphers, std::vector<Cipher> *toBeMovedCiphers);
+
+  static void moveToStart(std::vector<Cipher> *ciphers, std::vector<Cipher> *toBeMovedCiphers);
+  static void add(std::map<std::string, std::vector<Cipher>> *aliases, std::vector<Cipher> *ciphers,
+                  std::string alias);
+
+  static void remove(std::map<std::string, std::vector<Cipher>> *aliases,
+                     std::vector<Cipher> *ciphers, std::string alias);
+
+  static void strengthSort(std::vector<Cipher> *ciphers);
+
+  /*
+   * See
+   * https://github.com/openssl/openssl/blob/7c96dbcdab959fef74c4caae63cdebaa354ab252/ssl/ssl_ciph.c#L1371
+   */
+  static std::vector<Cipher> defaultSort(std::vector<Cipher> *ciphers);
```

Comment:
I would make all these vector arguments into references, unless you have a need for `nullptr` arguments.
```suggestion
  static std::vector<Cipher> defaultSort(std::vector<Cipher> &ciphers);
```

Reply:
Definitely agree 👍